cpp/tests/lists/set_operations/difference_distinct_tests.cpp

```diff
@@ -241,13 +241,19 @@ TEST_F(SetDifferenceTest, StringTestsWithNullsEqual)
        strings_lists{}, /* NULL */
        strings_lists{"aha", "this", "is another", "string???"}},
       null_at(1)};
-    auto const expected = strings_lists{{strings_lists{"a", "is", "string", "this"},
-                                         strings_lists{} /*NULL*/,
-                                         strings_lists{"a", "is", "string"}},
-                                        null_at(1)};
+    auto const expected = [] {
+      auto str_lists = strings_lists{{strings_lists{"a", "is", "string", "this"},
+                                      strings_lists{} /*NULL*/,
+                                      strings_lists{"a", "is", "string"}},
+                                     null_at(1)}
+                         .release();
+      auto& child = str_lists->child(cudf::lists_column_view::child_column_index);
+      child.set_null_mask(cudf::create_null_mask(child.size(), cudf::mask_state::ALL_VALID), 0);
+      return str_lists;
+    }();
 
     auto const results_sorted = set_difference_sorted(lhs, rhs, NULL_EQUAL);
-    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected, *results_sorted);
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(*expected, *results_sorted);
```

Comment:
I realize this is a test that will terminate very soon, so it's not a big deal, but the changes here introduced a memory leak. Why call `release` above, why not just `std::move` the output if RVO isn't happening for you (although I think it should?)?

Reply:
`strings_lists` here is a `column_wrapper`, so `release()` will return a `unique_ptr<column>`. I guess you're misunderstanding it as `release()` is releasing the raw pointer.

We need it have `expected` as a `unique_ptr<column>` to access the `column& column::child()` function, which ultimately gives us the ability to call `set_null_mask`.

Reply:
You're absolutely right, I thought it was a column and missed that it was a wrapper.